### PR TITLE
update with-linaria example

### PR DESCRIPTION
**Pull request by turadg**
## Documentation / Examples

* removes deprecated `@zeit/next-css`.
* adds https://github.com/Mistereo/next-linaria to work with the new built-in CSS functionality.
* updates Linaria to latest release (v3 will require https://github.com/Mistereo/next-linaria/pull/3 ) 
* adds an example of using Linaria without React
* gitignores `.linaria-cache`

### CODE REVIEW

File: examples/with-linaria/package.json [hunk 1]
```diff
@@ -7,9 +7,10 @@
     "start": "next start"
   },
   "dependencies": {
-    "@zeit/next-css": "^1.0.1",
-    "linaria": "2.0.0-alpha.5",
+    "@babel/core": "^7.13.10",
```
Comment (leerob) [inline examples/with-linaria/package.json:10]: Is babel needed here?

Comment (lfades) [inline examples/with-linaria/package.json:10]: Why is this dependency required? doesn't seem to be needed in the readme of [next-linaria](https://github.com/Mistereo/next-linaria)

Comment (turadg) [reply examples/with-linaria/package.json:10]: Yep, it's a peer dependency of `linaria`. Without it the build errors,
```
error - ./pages/index.js
Error: Cannot find module '@babel/core'
Require stack:
- /Users/turadg/Code/next.js/examples/with-linaria/node_modules/linaria/lib/transform.js
- /Users/turadg/Code/next.js/examples/with-linaria/node_modules/linaria/lib/loader.js
```

Comment (turadg) [reply examples/with-linaria/package.json:10]: Yeah, it's not required by `next-linaria` but by `linaria` itself. https://github.com/vercel/next.js/pull/23332#discussion_r600013278

Comment (lfades) [reply examples/with-linaria/package.json:10]: Can you add it as a `devDependency` 🙏

**Review by lfades — DISMISSED** [review]
Thank you!

**Review by leerob — APPROVED** [review]

**Pull request merged**